src/js/api/profile/fetchPosts.mjs

```diff
@@ -0,0 +1,34 @@
+// fetchPosts.mjs
+import { API_SOCIAL } from "../../constants.mjs";
+import { getLocal } from "../../storage/storage.mjs";
+
+export async function fetchUserPosts() {
+  const userData = getLocal("userData");
+  if (!userData || !userData.name) {
+    console.error("No user data available.");
+    return [];
+  }
+
+  const url = `${API_SOCIAL}/profiles/${encodeURIComponent(
+    userData.name
+  )}/posts`;
+
+  try {
+    const response = await fetch(url, {
+      headers: {
+        Authorization: `Bearer ${userData.accessToken}`,
+        "Content-Type": "application/json",
+      },
```

Comment:
Could benefit from a reusable headers function, as i see you type it up in all api calls.